system76/default.nix

```diff
@@ -0,0 +1,72 @@
+# Implementation of support for System76 hardware, controllable by NixOS options
+# named hardware.system76.*.
+
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.hardware.system76;
+
+  useDkms = cfg.drivers.system76.enable;
+  useAcpiDkms = cfg.drivers.system76-acpi.enable;
+  useIoDkms = cfg.drivers.system76-io.enable;
+
+  kernelPackages = let inherit (config.boot.kernelPackages) callPackage; in {
+    system76 = (callPackage pkgs/system76-dkms {});
+    system76-acpi = (callPackage pkgs/system76-acpi-dkms {});
+    system76-io = (callPackage pkgs/system76-io-dkms {});
```

Comment:
Could you add those packages to nixpkgs itself? We don't have the infrastructure in this repository to test builds and this way it would also be cached by hydra.
If you link the PR here, I will review it.

Reply:
Good point, the packages belong over there.  PR created: https://github.com/NixOS/nixpkgs/pull/96514